### DISCUSSION TITLE
API hosting on vercel platform ?

### DISCUSSION THREAD

**Discussion by remcoz** [vercel/community#6004] [Feature Question]
### Question

We use vercel for our FE app (nuxt).   Our API (rails) running on servers on other provider.  Can host API server on vercel platform?

**Comment by xonx4l** [comment]
Yes, you can host a Ruby on Rails API application on Vercel. Here are a few tips for doing so:

Vercel supports building and deploying Ruby apps. You'll need to have a vercel.json configuration file that specifies the build commands and output directory.

For a Rails API, your vercel.json would look something like this to build the app and output it to the public directory:

{
  "builds": [
    {
      "src": "api/",
      "use": "@vercel/ruby",
      "config": { "command": "rails api:build" } 
    }
  ],
  "routes": [
    { "src": "/(.*)", "dest": "api/public/$1" }
  ]
}



You can deploy a PostgreSQL database add-on through Vercel to provide a database for your Rails API.

Make sure your Rails config is set for production environment on Vercel, and not using a local database config.

You'll need to configure environment variables on Vercel for any credentials, secrets, etc your API needs.

Consider putting a CDN like Vercel Edge Functions in front of your API endpoints for caching, rate limiting, etc.

So in summary - yes Vercel can deploy a Rails API app, you just need to configure it properly for the environment and provide the required dependencies like a database
